app/invites/server/functions/findOrCreateInvite.js

```diff
@@ -0,0 +1,105 @@
+import { Meteor } from 'meteor/meteor';
+import { Random } from 'meteor/random';
+
+import { hasPermission } from '../../../authorization';
+import { Notifications } from '../../../notifications';
+import { Invites, Rooms } from '../../../models';
+import { settings } from '../../../settings';
+
+function getInviteUrl(invite) {
+	const { rid, hash } = invite;
+
+	const useDirectLink = settings.get('Accounts_Registration_InviteUrlType') === 'direct';
+	// Remove the last dash if present
+	const siteUrl = settings.get('Site_Url').replace(/\/$/g, '');
```

Comment:
Could use the `getUrl` method from [here](https://github.com/RocketChat/Rocket.Chat/blob/919a55ea5c446d4356548e7fdd5957eaf7685a53/app/utils/lib/getURL.js#L7) and create an extra option there `redirectViaCloud` that would populate the `host` param and will receive others via an object like `redirectViCloudParams` so we can reuse this logic.

Reply:
What about this review?